x-pack/plugins/lens/public/indexpattern_datasource/datapanel.test.tsx

```diff
@@ -856,24 +860,26 @@ describe('IndexPattern Data Panel', () => {
         );
         const wrapper = mountWithIntl(<InnerIndexPatternDataPanel {...props} />);
         act(() => {
-          (
-            wrapper
-              .find('[data-test-subj="lnsIndexPatternActions-popover"]')
-              .first()
-              .prop('children') as ReactElement
-          ).props.items[0].props.onClick();
+          const popoverTrigger = wrapper.find(
+            '[data-test-subj="lnsIndexPatternActions-popover"] button'
+          );
+          popoverTrigger.simulate('click');
         });
 
+        wrapper.update();
+        act(() => {
+          wrapper.find('[data-test-subj="indexPattern-add-field"]').first().simulate('click');
+        });
         // wait for indx pattern to be loaded
-        await act(async () => await new Promise((r) => setTimeout(r, 0)));
```

Comment:
I am a bit on the edge of replacing this with `waitFor` - I think it's more readable, but it comes from react-testing-library and we use in 99% enzyme. There are couple of more places where we use react-testing-library in lens (eg. testing debouncing hook) and some other teams use it extensively in Kibana, so I am not sure. Thoughts?

Reply:
The behaviour is different between the 2 things.
I think `waitFor` is more reliable than a manual arbitrary waiting. So 👍